tools/test-js-with-node

```diff
@@ -155,7 +173,21 @@ print('Starting node tests...')
 
 # If we got this far, we can run the tests!
 try:
-    ret = subprocess.check_call(command)
+    # This is duplicate code of:
+    #   `def run_parallel(lint_functions)` via commands.py
+    # A possible TODO: Refactor the function to work for both cases.
+    pids = []
+    for tests in sub_tests:
+        pid = os.fork()
+        if pid == 0:
+            os.execvp('node', command + tests)
+        pids.append(pid)
+    ret = 0
+
+    for pid in pids:
+        (_, status) = os.waitpid(pid, 0)
+        if status != 0:
+            ret = status
```

Comment:
Shouldn't something be done with the `ret` in case status wasn't `0`?

Reply:
Yeah, that's handled at the end (if this is what you're referring to):
```
if ret == 0:
    if options.coverage:
        print("View coverage reports at http://127.0.0.1:9991/node-coverage/index.html")
    print("Test(s) passed. SUCCESS!")
else:
    print("FAIL - Test(s) failed")

sys.exit(ret)
```

Reply:
ahh, okay. I didn't see that at the bottom. Code is itself structured a bit strangely such that after the call to the tests there are some function definitions which are required for the coverage code path and after that we have checks for the returned status code.